### PR TITLE
Fix word repetion in all the docs

### PR DESCRIPTION
**Pull request by tetedacier**
Those has been found thanks to the execution accross all the docs of
this rudimentary regex: `\b([^\s\d]{2,5})\s+\1\b`

It is inspired from the the one described by Jeffrey Friedl in his book
`Mastering Regular Expression`.

### CODE REVIEW

**Comment by frank-trampe** [comment]
@tetedacier, thanks for doing this. I've marked up a few details that seemed off. Before you do anything else, though, I want to have a quick discussion about which pages we consider historical monuments rather than working documents. @ctrlcctrlv, what do you think?

**Comment by jtanx** [comment]
This is probably better fixed in the fontforge repo itself, the non-archived stuff will be overwritten the next time we push updates back here

**Comment by frank-trampe** [comment]
@jtanx, good catch!

File: archive/ff-history.html [hunk 1]
```diff
@@ -446,10 +446,10 @@ <H2>
   svg) fonts.
   <P>
   In June I was thinking of the embolden command I did the year before, and
-  realized that that was essentially the same idea as was needed for generating
+  realized that was essentially the same idea as was needed for generating
```
Comment (ctrlcctrlv) [inline archive/ff-history.html:449]: Word repetition is not always incorrect grammatically. In fact, the new line here makes no sense. All changes must be read by a native speaker and verified, especially places "that" is duplicated.

File: archive/ff-history.html [hunk 2]
```diff
@@ -446,10 +446,10 @@ <H2>
...
   Small Caps glyphs from Capital letters. And then some of those algorithms
   could be used to create condensed and extended glyphs. And then I sat down
-  and wrote a generic "glyph change" dialog -- years ago I had had a "MetaFont"
+  and wrote a generic "glyph change" dialog -- years ago I had a "MetaFont"
```
Comment (tshinnic) [inline archive/ff-history.html:452]: Same as above, it is not non-standard English to repeat 'had'

File: archive/oldchangelog.html [hunk 3]
```diff
@@ -208,7 +208,7 @@
 	  table. (Needed because of a bug which introduced garbage).
 	<LI>
 	  We didn't think we could read in a kerning subtable (class or pair) with
-	  device tables, so we converted them to pst data. We can read them in in optimized
+	  device tables, so we converted them to pst data. We can read them in optimized
```
Comment (tshinnic) [inline archive/oldchangelog.html:211]: Can be read as akin to "We can read them in(to memory) in optimized form ..." These constructs may be seen less often now, but were not non-standard.

And if I'm reading the filename correctly, you are trying to edit an old changelog? *Really*!?? Methinks 'automatic' is overstated here?

File: archive/oldchangelog.html [hunk 4]
```diff
@@ -3151,7 +3151,7 @@
 	  synchronized to dependent glyphs.
 	<LI>
 	  Enforce the "Use My Metrics" bit of a reference.<BR>
-	  In TrueType, if this bit is set on a reference then then width of the containing
+	  In TrueType, if this bit is set on a reference then width of the containing
```
Comment (tshinnic) [inline archive/oldchangelog.html:3154]: Yes, this should have been "then the", but really, skip this file at least. And again, blunt application of tools is not indicated where human languages are employed.

File: archive/prefs.html [hunk 5]
```diff
@@ -147,7 +147,7 @@ <H2>
       "On" because it does not get in the way and will ensure that the
       correct extension is given to your font files as you work on
       them. This makes it much harder to accidentally ship a
-      Fontforge SFD file as as binary file or try to use an SFD format
+      Fontforge SFD file as binary file or try to use an SFD format
```
Comment (tshinnic) [inline archive/prefs.html:150]: Probably was supposed to have been "as a binary file", yes?

File: archive/scripting.html [hunk 6]
```diff
@@ -1234,7 +1234,7 @@ <H3 ALIGN=Center>
     <LI>
       <A HREF="scripting-alpha.html#SetPref">SetPref(str,val[,val2])</A>
     <LI>
-      It it also possible to get the value of a preference item by preceding its
+      It is also possible to get the value of a preference item by preceding its
```
Comment (tshinnic) [inline archive/scripting.html:1237]: Yes, exactly. Use the output of the 'tool' to direct attention towards *possible* problems, and start building changes upon evaluation.

Thus "had had" and "that that" are dropped as merely stylistic quirks. And other patterns emerge that are false positives. 

Use the tools, don't let the tools use you!

(Sorry, too many years at Wikipedia cleaning up after naïve editors' enthusiasms with 'tools'. Dozens and hundreds of clean up actions tell me tools have sharp edges, apparently unexpectedly)

File: docs/index.html [hunk 7]
```diff
@@ -473,7 +473,7 @@ <h3>What is Apple Advanced Typography?<a class="headerlink" href="#what-is-apple
 reference) will bring up a dialog showing you what glyph is referred to, and
 allowing you to bring up an editing window on that glyph. Finally the
 <a class="reference internal" href="ui/menus/elementmenu.html#elementmenu-accented"><span class="std std-ref">Element-&gt;Build-&gt;Accented Glyphs</span></a> command will
-figure out what glyphs should be refered to to build this composite glyph, and
+figure out what glyphs should be refered to build this composite glyph, and
```
Comment (tshinnic) [inline docs/index.html:476]: This also was correct. But if you have to change something, how about the misspelled word? Didn't notice that?

File: docs/old/de/editexample6.html [hunk 8]
```diff
@@ -164,7 +164,7 @@ <H2>
     erlaubt die Eingabe dieser Information.
   <P>
   Dann wird in jedem Zeichen, daß Anhänge besitzen soll, zunächst die
-  Stelle angeklickt, an der der Anker erstellt werden soll und dann der
+  Stelle angeklickt, an der Anker erstellt werden soll und dann der
```
Comment (tshinnic) [inline docs/old/de/editexample6.html:167]: My rudimentary acquaintance with Deutsch seems to agree with gTranslate that this is not a repeated word. Second 'der' is the article for "the Anchor". First 'der' is sorta like the word 'which' or 'what'. Thus "at *which* *the* anchor"

File: docs/old/ja/charview.html [hunk 9]
```diff
@@ -790,7 +790,7 @@ <H2>
 <!--
   The image to the right shows an example of this mode. You invoke it with
   <CODE><A HREF="hintsmenu.html#Debug">Hints-&gt;Debug</A></CODE>, and as with
-  the grid-fit view above you must establish a pointsize and and resolution.
+  the grid-fit view above you must establish a pointsize and resolution.
```
Comment (tshinnic) [inline docs/old/ja/charview.html:793]: Could have been intended to be "and a". Also, this is inside of an HTML comment???

**Comment by tshinnic** [comment]
Sorry, and I'll stop with the comments. I think I've demonstrated that while many of these proposed changed are valid, about 1 in 4 or 5 are not. Not noticing that some of these files were changelogs is disturbing. Reaching into non-English texts where the uncertainty is multiplied, likewise. And HTML comments... zounds!

How could this have been better accomplished, starting from a raw unreviewed and dangerous output?

**Comment by jtanx** [comment]
Some good points raised, but I'm closing this to avoid further confusion - the documentation is generated from Sphinx/rst files in the main fontforge/fontforge repo, and I don't think it's worth bothering to update the archived documentation at this point (it's archived!!!).

If you'd like to make improvements, please update the files under https://github.com/fontforge/fontforge/tree/master/doc/sphinx instead.

File: archive/ff-history.html (hunk 2 above)
Comment (JoesCat) [reply archive/ff-history.html:452]: Reminded me of a riddle....
@tshinnic, while @tetedacier had had 'had', had had 'had had'. 'had had' had had a better effect in the file.

File: docs/old/de/editexample6.html (hunk 8 above)
Comment (lemzwerg) [reply docs/old/de/editexample6.html:167]: @tshinnic, you are correct.  There have to be two "der".

**Comment by tetedacier** [comment]
Firstly, I’d like to thanks all of you for all your reviews, I learned a lot from all of them.

By the way, I’ve made the mistake described in this PR title in the second sentence of my comment :man_facepalming:
lack of sleep makes me weak and less focused …

@tshinnic: I did’t apply my regex automatically, it was just a guide, I still have 1555 matches of this pattern which mainly concern html class attributes :smile: another reason for my lack of focus on the meaning of impacted sentences.

@jtanx: I get your point, and will back port relevant and reviewed change in `fontforge` main repository if nobody has already done it.

@jtanx, @tshinnic: I’m thinking about a smart way to handle this `archive` issue (lots of old folder are here for inner archive references) directly in git. I will let you know in the issues tracker.

Cheers !

**Pull request closed without merging**